### PR TITLE
Terminate pre-flight requests

### PR DESCRIPTION
**Pull request by ryanbrunner**
When cors-gate receives a preflight request, it should terminate that request successfully
rather than forwarding the request down to other middleware.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+ * Called when we successfully gate a request. If the
+ * request is an OPTION request, terminate the request here.
+ * Otherwise, pass on to the next middleware.
+ */
+function success(req, res, next) {
+  if (req.method === 'OPTIONS') {
+    // If this is an OPTIONS request, terminate here.
+    res.statusCode = 204;
+    // Safari needs a content-length for 204, see https://github.com/expressjs/cors/blob/master/lib/index.js#L176
+    res.setHeader('Content-Length', '0');
```
Comment (skeggse) [inline index.js]: Oh nice catch!

**Comment by ryanbrunner** [comment]
@skeggse do you think we need to add anything to the README for this? It *should* be something that's OK for any use case I can think of, but you never know.

**Comment by skeggse** [comment]
I think it's fine to leave as-is. I also think it can't hurt to add more docs ;)

**Comment by wearhere** [comment]
Was this always a bug? Or is this caused by some other recent change?

File: test/cors-gate.spec.js [hunk 2]
```diff
@@ -71,6 +71,21 @@ describe('cors-gate', function() {
       .expect(200, done);
   });
 
+  it('should terminate on preflight requests', function(done) {
```
Comment (wearhere) [inline test/cors-gate.spec.js]: thanks for adding tests!

File: test/cors-gate.spec.js [hunk 3]
```diff
@@ -71,6 +71,21 @@ describe('cors-gate', function() {
...
+    }), corsGate({
+      origin: 'http://localhost'
+    }));
+
+    this.app.post('/post', ok);
+
+    request(this.app)
+      .options('/post')
+      .set('origin', 'http://localhost:8080')
+      .expect(204, done);
```
Comment (wearhere) [inline test/cors-gate.spec.js]: Could you also test the zero content length? Since that's important external behavior

**Comment by ryanbrunner** [comment]
@wearhere re: "was this always a bug". That depends on your definition :) We previously relied on the `cors` module from npm terminating OPTIONS request (which our changes to `cors` removed), but when talking with @skeggse we thought it made more sense to terminate `OPTIONS` requests here to minimize the surface area of invalid requests (we'd rather that a preflight request fail if it's not valid than say everything is OK since it never reached `cors-gate`.

For non-Mixmax users who are using this with the `cors` module this change should have zero effect (since preflight requests will be terminated prior to this). If anyone is not terminating their preflight requests though, this will terminate them.

**Comment by ryanbrunner** [comment]
@wearhere Merging this so that I can move forward on testing the fix, but if you have any concerns about the last comment, feel free to DM me.

**Comment by wearhere** [comment]
@ryanbrunner all sounds good! Nice that you considered non-Mixmax users.
